### PR TITLE
fix: Length property in logs endpoint is a string

### PR DESCRIPTION
**Pull request by danielgent**
image

### CODE REVIEW

File: src/adapters/in-memory/logs/list.ts [hunk 1]
```diff
@@ -9,6 +9,10 @@ export function listLogs(logs: Log[]) {
   ): Promise<ListLogsResponse> => {
     return {
       logs,
+      // no pagination just return all
+      start: 0,
+      length: logs.length,
+      limit: logs.length,
```
Comment (danielgent) [inline src/adapters/in-memory/logs/list.ts:15]: We're no longer type casting with `as` so the type errors come straight through from typescript :sunglasses:

File: src/adapters/interfaces/Logs.ts [hunk 2]
```diff
@@ -9,9 +9,8 @@ export interface CreateLogParams {
   user_id: string;
 }
 
-export interface ListLogsResponse {
+export interface ListLogsResponse extends Totals {
```
Comment (danielgent) [inline src/adapters/interfaces/Logs.ts:12]: this is what the user endpoint does and what we wer kind of doing (but hidden with the type casting)

File: src/adapters/planetscale/logs/list.ts [hunk 3]
```diff
@@ -31,21 +32,17 @@ export function listLogs(db: Kysely<Database>) {
...
       .execute();
 
+    const countInt = getCountAsInt(count);
+
     return {
       logs,
       start: (params.page - 1) * params.per_page,
       limit: params.per_page,
-      length: count,
+      length: countInt,
```
Comment (danielgent) [inline src/adapters/planetscale/logs/list.ts:45]: the fix that the ticket asks for :muscle:

File: src/adapters/planetscale/tenants/listTenants.ts [hunk 4]
```diff
@@ -1,15 +1,7 @@
...
 import { Kysely } from "kysely";
 import { ListParams } from "../../../adapters/interfaces/ListParams";
-
-function getCountAsInt(count: string | number | bigint) {
-  // VScode complains that parseInt only accepts a string BUT the project builds & lints
-  if (typeof count === "string") {
-    return parseInt(count, 10);
-  }
-
-  return count;
```
Comment (danielgent) [inline src/adapters/planetscale/tenants/listTenants.ts:11]: Cleaning up duplicated code

File: src/adapters/planetscale/users/list.ts [hunk 5]
```diff
@@ -101,12 +88,6 @@ export function listUsers(db: Kysely<Database>) {
 
     const users = await filteredQuery.selectAll().execute();
 
-    // if (!params.include_totals) {
-    //   return {
-    //     users,
-    //   };
-    // }
```
Comment (danielgent) [inline src/adapters/planetscale/users/list.ts:108]: handling this in the actual route

File: src/routes/management-api/logs.ts [hunk 6]
```diff
@@ -46,7 +46,13 @@ export class LogsController extends Controller {
     });
 
     if (include_totals) {
-      return result as ListLogsResponse;
+      const res: ListLogsResponse = {
```
Comment (danielgent) [inline src/routes/management-api/logs.ts:49]: :smile:

File: src/adapters/in-memory/logs/list.ts (hunk 1 above)
Comment (markusahlstrand) [reply src/adapters/in-memory/logs/list.ts:15]: Think we want to add the pagination here pretty soon :)

File: src/routes/management-api/logs.ts (hunk 6 above)
Comment (markusahlstrand) [reply src/routes/management-api/logs.ts:49]: We should have a lint rule for the `as` :)

**Review by markusahlstrand — APPROVED** [review]

**Pull request merged**